tests/postgres_tests/test_hstore.py

```diff
@@ -208,6 +208,13 @@ def test_invalid_json(self):
         self.assertEqual(cm.exception.messages[0], 'Could not load JSON data.')
         self.assertEqual(cm.exception.code, 'invalid_json')
 
+    def test_badly_formatted_json(self):
+        field = forms.HStoreField()
+        with self.assertRaises(exceptions.ValidationError) as cm:
+            field.clean('["a", "b", 1]')
+            self.assertEqual(cm.exception.messages[0], 'Input should resemble a dictionary.')
```

Comment:
```python
msg = 'Input should resemble a dictionary.'
with self.assertRaisesMessage(exceptions.ValidationError, msg) as cm:
    field.clean('["a", "b", 1]')
self.assertEqual(cm.exception.code, 'invalid_format')
```

Note that the assertions must be run outside the context manager else they will never be executed as the `ValidationError` raised by `clean()` will force an immediate `__exit__()`.